flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequest.java

```diff
@@ -98,6 +98,9 @@ static ChannelStateWriteRequest buildFutureWriteRequest(
                     List<Buffer> buffers;
                     try {
                         buffers = dataFuture.get();
+                    } catch (InterruptedException e) {
+                        writer.fail(e);
+                        throw e;
```

Comment:
How do you think it's related to FLINK-27792?

Reply:
I'm interpreting that ticket as a more general "how to handle InterruptedException" ticket.

Reply:
Hi @zentol @pnowojski , I replied in [FLINK-27792](https://issues.apache.org/jira/browse/FLINK-27792). And I think the root cause of FLINK-27792 is we don't have a reasonable cleanup dataFuture in discardAction, so we should improve the discardAction.

I improved action in this change because it is also a bug, we should continue throw InterruptedException here. 


And I think this change and [FLINK-27792](https://issues.apache.org/jira/browse/FLINK-27792) are different. And I think the [FLINK-27792](https://issues.apache.org/jira/browse/FLINK-27792) is the same root cause as [FLINK-28077](https://issues.apache.org/jira/browse/FLINK-28077), that is next change in this PR. Defective discardAction resulted in FLINK-27792 and FLINK-28077. Once this PR is done, they will be resolved.

I know next change still has some bugs, it may cause network memory leak. I will think more and think about how to solve it. What do you think? If anything is wrong, please correct me.

Reply:
Hi @zentol @pnowojski , after I analyzed, I found we don't need change this. 

This change compared to before just calling `writer.fail(e);` when InterruptedException is caught. The previous code did not catch InterruptedException, so it will continue to throw when InterruptedException is encountered.

We don't need call the `writer.fail(e);` when InterruptedException is caught. Because the `dispatcher.fail()` will be called before ChannelStateWriteThread dead. [code link](https://github.com/apache/flink/blob/3ae4c6f5a48105d00807e8ce02e70d4c092cbf40/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorImpl.java#L83). 

The dispatcher.fail will call all writers's fail() method, so I think these change isn't necessary. I will revert it in next commit. We will focus on next change, that is, improve the discardAction.